src/error.rs

```diff
@@ -309,11 +309,11 @@ impl<I: fmt::Display + Hash + Eq, S: Span> fmt::Display for Simple<I, S> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         // TODO: Take `self.reason` into account
 
-        if let Some(found) = &self.found {
-            write!(f, "found '{}'", found)?;
+        let found = if let Some(found) = &self.found {
+            format!("{}", found)
         } else {
-            write!(f, "found end of input")?;
-        }
+            format!("found end of input")
```

Comment:
```suggestion
            format!("end of input")
```
otherwise you'd get `found found end of input ...`

Reply:
Definitely true. I agree with this change. I overlooked that part.